src/compiler/crystal/codegen/llvm_builder_helper.cr

```diff
@@ -30,6 +30,27 @@ module Crystal
       int32(n)
     end
 
+    def size_t
+      llvm_context.int(@program.size_bit_width)
+    end
+
+    def size_t(n)
+      size_t.const_int(n)
+    end
+
+    def size_t(value : LLVM::Value)
+      case value.type.int_width <=> @program.size_bit_width
+      when .zero?
+        value
+      when .positive?
+        builder.trunc(value, size_t)
+      when .negative?
+        builder.zext(value, size_t)
+      else
+        raise "unreachable"
+      end
```

Comment:
**polish:** Could we drop this last condition and go directly into an `else` branch if the comparison is neither zero nor negative?
```suggestion
      else
        builder.zext(value, size_t)
      end
```